gef-ui/server/services/api.js

```diff
@@ -113,7 +113,7 @@ const setApplicationStatus = async ({ dealId, status, userToken }) => {
       {
         status,
       },
-      { ...config(userToken), timeout: 10000 },
```

Comment:
timeout required as sometimes this takes just over 10 seconds (at least on my local build, wouldn't be good if this only worked on prod). Without this the user sees a "something wrong with the service page" even though there actually isn't, we've just got a lot of chained API calls on submission, to which this PR adds one. According to the root README:

> When a deal is submitted to TFM, many external API calls are currently made in the TFM submission controller. This process can be slow, resource-intensive, and prone to failures if one of the API calls encounters an issue. Retries are not currently configured.

> To address these issues, the plan is to move all these API calls into background processes with automatic retries. This will enhance the user experience, make the process fail-safe, and improve the development workflow.

However, that work isn't in the scope of this PR of course

Reply:
Okay makes sense, I think given we are already close to the timeout can we make this 20 seconds instead of 30? We might want to add some extra documentatoin in the form of a JS-doc to the top of this function explaining that this goes off to do lots of different (long) API calls and that's why the timeout is very long

Reply:
The timeout increase concerns me - how long does deal submission currently take with a production build?

I am wondering if we should create a CRON job for this functionality instead of increasing submission time.

Depending on the timings and if the increase isn't that bad, perhaps this could be something we could look into as a future iteration (maybe an "improve submission time/performance" epic pending business priorities etc.

Reply:
I think there's a plan to move the deal submission to a background task long-term, and I agree that's a good idea. I'm not sure if falls in the scope of this PR though, all the deal submission tasks should be included in that, not just this one. I'm not sure how long this will take on a production build, though I'd hope it would come in under 10 seconds (it's only slightly over on my local build).

Reply:
I agree with @ChristianMcc  we should increase this to `30,000ms` for fail-safe and in my current PR which tackles this issue I will speak to the UCD to temporarily show a processing page.

Long-term I would refer to Azure functions like approach where multiple external API calls can be handled and if either fails an alert is dispatched and reverting the deal status.

Thanks @natdeanlewissoftwire for a precise and concise explanation and agree with @ChristianMcc  if above can be added to DocString please.